### ISSUE TITLE
Should compressor ignore external resource instead of throwing error?

### ISSUE THREAD

**Issue by jproffitt** [django-compressor/django-compressor#585]
When you add an external resource inside a `{% compress %}` block, it fails hard saying it "isn't accessible via `COMPRESS_URL`". Would it be better to not try to compress it, and just print it out like normal?

Normally this isn't a big deal, as you could simply move it outside of the compress tag, but when you use compressor with sekizai, it's a bit of a pain.

Here is my use case:

I use sekizai like this in my base template:

```
{% render_block 'js' postprocessor "compressor.contrib.sekizai.compress" %}
```

I might add some scripts like this:

```
{% addtoblock 'js' %}<script type="text/javascript" src="{% static 'js/jquery.min.js' %}"></script>{% endaddtoblock %}
```

Then, in another template, I would like to add an external script from a CDN. So I would normally do this:

```
{% addtoblock 'js' %}<script type="text/javascript" src="https://js.stripe.com/v2/"></script>{% endaddtoblock %}
```

But then I get the error saying that the stripe script is not available via `COMPRESS_URL`.

It would be great if it could just continue compressing everything else, and spit out stripe like normal:

```
<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
<script type="text/javascript" src="/static/CACHE/js/5e03a114934d.js"></script>
```

I guess the big problem would be how it would figure out the order. Maybe it should make separate compressed files for anything before external files, and another one for anything after.

This could be an opt-in solution.

**Comment by sawall** [comment]
This seems like a really strange use case.

You would rather download a script from a CDN to Django, package it, and serve it yourself to to host rather than just let the host download it?

**Comment by jproffitt** [comment]
No, I would not rather do that. And thats the problem. If I use compressor with sekizai, I cannot have define external resources like from a CDN within my sekizai block.

I could, like @meisme said, just make a new sekizai block for external resources, and that is what I do now. But that can be a pain if you have an external resource that relys on a local resource, and then another local resource that relys on that external resource. I would have to have 3 blocks then. One for everything before the external resource, one for that external resource, and another for everything after it.

I'm just saying it would be nice if compressor could be smart, and figure all that out automatically. Right now, if it finds an external that is location somewhere other than `COMRESS_URL`, it fails hard. Would be nice if it would just not even try to compress it, and compress everything before it, and everything after it.

I hope I'm making sense here. Let me give another example:

base.html:

    {% render_block 'js' postprocessor "compressor.contrib.sekizai.compress" %}

other_page.html

    {% addtoblock 'js' %}<script type="text/javascript" src="{% static 'js/jquery.min.js' %}"></script>{% endaddtoblock %}
    {% addtoblock 'js' %}<script type="text/javascript" src="https://js.stripe.com/v2/"></script>{% endaddtoblock %}
    {% addtoblock 'js' %}<script type="text/javascript" src="{% static 'js/my_scripts.js' %}"></script>{% endaddtoblock %}

output would be:

    <script type="text/javascript" src="/static/CACHE/js/5e03a11491h6.js"></script>
    <script type="text/javascript" src="https://js.stripe.com/v2/"></script>
    <script type="text/javascript" src="/static/CACHE/js/5e03a114934d.js"></script>

That way I don't need 3 sekizai blocks.

**Comment by sinaso** [comment]
+1 on jproffitt request. This is specially needed if you have a a bunch of resources defined in Form Media and want to include all those into a compress block but some of those assets cannot be local, like stripe.js for payment processing. Would be nice if the default behavior is to just ignore remote resources and puts them in the output.

**Comment by Lacrymology** [comment]
:+1:

**Comment by karyon** [comment]
#891 made this work with sekizai, if i understand correctly. without sekizai, the answer is probably still to use multiple compress blocks.
